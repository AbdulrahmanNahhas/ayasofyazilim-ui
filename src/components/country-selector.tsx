'use client';

import { CaretSortIcon } from '@radix-ui/react-icons';
import * as React from 'react';
import Flag from 'react-world-flags';
import { Button } from '@/components/ui/button';
import {
  Command,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
  CommandList,
} from '@/components/ui/command';
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from '@/components/ui/popover';

type CountrySelectItem = {
  label?: string;
  rtl?: boolean;
  value?: string;
};
const countries = [
  {
    value: 'tur',
    label: 'Türkçe',
  },
  {
    value: 'ita',
    label: 'Italiano',
  },
  {
    value: 'jpn',
    label: '日本語',
  },
  {
    value: 'sau',
    label: 'عربي',
    rtl: true,
  },
];

export type CountrySelectorProps = {
  defaultValue: CountrySelectItem;
  direction: 'ltr' | 'rtl';
  menuAlign: 'start' | 'center' | 'end';
  onValueChange: () => {};
  searchEmptyValue: string;
  searchText: string;
};
export default function CountrySelector({
  searchText,
  searchEmptyValue,
  defaultValue,
  menuAlign,
  direction,
  onValueChange,
}: CountrySelectorProps) {
  const [open, setOpen] = React.useState(false);
  const [value, setValue] = React.useState('');

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button
          variant="outline"
          role="combobox"
          aria-expanded={open}
          className={`justify-between px-2 gap-2 ${direction === 'rtl' ? 'flex-row-reverse' : ''}`}
        >
          {value ? (
            <SelectedCountry
              {...countries.find((country) => country.value === value)}
            />
          ) : (
            <SelectedCountry {...defaultValue} />
          )}
          <CaretSortIcon className="h-4 w-4 shrink-0 opacity-50" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[100px] p-0" align={menuAlign}>
        <Command>
          <CommandList>
            <CommandInput placeholder={searchText} className="h-9 text-xs" />
            <CommandEmpty>{searchEmptyValue}</CommandEmpty>
            <CommandGroup>
              {countries.map((country) => (
                <CommandItem
                  key={country.value}
                  value={country.value}
                  onSelect={(currentValue) => {
                    setValue(currentValue);
                    setOpen(false);
                    onValueChange();
                  }}
                >
                  <SelectedCountry {...country} />
                </CommandItem>
              ))}
            </CommandGroup>
          </CommandList>
        </Command>
      </PopoverContent>
    </Popover>
  );
}

const SelectedCountry = ({ value, label, rtl }: CountrySelectItem) => (
  <div className={`${rtl ? 'flex-row-reverse' : ''} flex w-full gap-2`}>
    <Flag code={value} className="w-4" />
    <span className="text-xs">{label}</span>
  </div>
);
