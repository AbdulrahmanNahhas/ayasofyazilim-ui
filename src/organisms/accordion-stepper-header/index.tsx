'use client';

import { Circle, CircleCheckBig } from 'lucide-react';
import React from 'react';
import { AccordionTrigger } from '@/components/ui/accordion';

interface IAccordionStepperHeaderProps {
  checked: boolean;
  children: React.ReactNode;
}
export const AccordionStepperHeader = ({
  children,
  checked,
}: IAccordionStepperHeaderProps) => (
  <div className="bg-muted/50 px-4">
    <AccordionTrigger>
      <div className="flex gap-4 items-center">
        {checked ? (
          <CircleCheckBig className="text-primary" />
        ) : (
          <Circle className="text-muted-foreground" />
        )}
        {children}
      </div>
    </AccordionTrigger>
  </div>
);
