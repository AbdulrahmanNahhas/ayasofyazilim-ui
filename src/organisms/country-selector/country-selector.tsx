'use client';

import * as React from 'react';

import { Button } from '../../../@/components/ui/button';
import {
  Command,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
  CommandList,
} from '../../../@/components/ui/command';
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from '../../../@/components/ui/popover';

type CountrySelectItem = {
  label?: string;
  rtl?: boolean;
  showLabel?: boolean;
  value?: string;
};
const countries = [
  {
    value: 'tr',
    label: 'Türkçe',
  },
  {
    value: 'it',
    label: 'Italiano',
  },
  {
    value: 'jp',
    label: '日本語',
  },
  {
    value: 'sa',
    label: 'عربي',
    rtl: true,
  },
];

type CountrySelectorProps = {
  defaultValue?: CountrySelectItem;
  direction: 'ltr' | 'rtl';
  menuAlign: 'start' | 'center' | 'end';
  onValueChange?: () => {};
  searchEmptyValue?: string;
  searchText?: string;
};
function CountrySelector({
  searchText,
  searchEmptyValue,
  defaultValue,
  menuAlign,
  direction,
  onValueChange,
}: CountrySelectorProps) {
  const [open, setOpen] = React.useState(false);
  const [value, setValue] = React.useState<string | undefined>('');

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button
          role="combobox"
          variant="ghost"
          aria-expanded={open}
          className={`justify-between border-none bg-transparent px-2 gap-2 ${direction === 'rtl' ? 'flex-row-reverse' : ''}`}
        >
          {value ? (
            <SelectedCountry
              {...countries.find((country) => country.value === value)}
              showLabel={false}
            />
          ) : (
            <SelectedCountry {...defaultValue} showLabel={false} />
          )}
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[160px] p-0" align={menuAlign}>
        <Command>
          <CommandList>
            <CommandInput placeholder={searchText} className="h-9 text-xs" />
            <CommandEmpty>{searchEmptyValue}</CommandEmpty>
            <CommandGroup>
              {countries.map((country) => (
                <CommandItem
                  key={country.value}
                  value={country.value}
                  onSelect={(currentValue: string) => {
                    setValue(currentValue);
                    setOpen(false);
                    if (onValueChange) onValueChange();
                  }}
                >
                  <SelectedCountry {...country} />
                </CommandItem>
              ))}
            </CommandGroup>
          </CommandList>
        </Command>
      </PopoverContent>
    </Popover>
  );
}

const SelectedCountry = ({
  value = '',
  label,
  rtl,
  showLabel = true,
}: CountrySelectItem) => (
  <div
    className={`${rtl ? 'flex-row-reverse' : ''}  flex w-full justify-between gap-2 overflow-hidden`}
  >
    {showLabel && <span className="text-xs">{label}</span>}
    <div className="border  rounded-sm overflow-hidden">
      <img
        className="w-6"
        src={`https://cdn.jsdelivr.net/gh/lipis/flag-icons/flags/4x3/${value}.svg`}
        alt={label}
      />
    </div>
  </div>
);

export { CountrySelectItem, CountrySelector };
