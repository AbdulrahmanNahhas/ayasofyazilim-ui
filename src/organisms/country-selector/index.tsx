'use client';

import * as React from 'react';

import { Button } from '@/components/ui/button';
import {
  Command,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
  CommandList,
} from '@/components/ui/command';
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from '@/components/ui/popover';

type CountryItem = {
  concurrencyStamp?: string;
  creationTime?: string;
  creatorId?: string | null;
  cultureName?: string | null;
  direction?: 'rtl' | 'ltr' | null;
  displayName?: string | null;
  extraProperties?: object | null;
  flagIcon?: string | null;
  id?: string | null;
  isDefaultLanguage?: boolean;
  isEnabled?: boolean;
  twoLetterISOLanguageName?: string | null;
  uiCultureName?: string | null;
};

type CountrySelectorProps = {
  countries?: Array<CountryItem>;
  defaultValue?: string;
  menuAlign?: 'start' | 'center' | 'end';
  onValueChange?: (value: string) => void;
  searchEmptyValue?: string;
  searchText?: string;
  showLabel?: boolean;
  tooltipText?: string;
};

function CountrySelector({
  searchText,
  searchEmptyValue,
  defaultValue,
  menuAlign = 'end',
  showLabel = false,
  countries = [],
  onValueChange,
}: CountrySelectorProps) {
  const [open, setOpen] = React.useState(false);
  const [value, setValue] = React.useState<string>('');

  function onSelect(currentValue: string) {
    setValue(currentValue);
    setOpen(false);
    if (onValueChange) onValueChange(currentValue);
  }
  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger>
        <Button
          asChild
          role="combobox"
          variant="ghost"
          aria-expanded={open}
          className="justify-between border-none bg-transparent px-2 gap-2 rtl:flex-row-reverse min-h-8 bg-red-300"
        >
          {value ? (
            <SelectedCountry
              {...countries.find(
                (country) => country.cultureName?.toLowerCase() === value
              )}
              showLabel={showLabel}
            />
          ) : (
            <SelectedCountry
              {...countries.find(
                (country) => country.cultureName?.toLowerCase() === defaultValue
              )}
              showLabel={showLabel}
            />
          )}
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[160px] p-0" align={menuAlign}>
        <Command>
          <CommandList>
            <CommandInput placeholder={searchText} className="h-9 text-xs" />
            <CommandEmpty>{searchEmptyValue}</CommandEmpty>
            <CommandGroup>
              {countries.map((country) => (
                <CommandItem
                  key={country.cultureName}
                  value={country.cultureName || ''}
                  onSelect={(currentValue: string) => onSelect(currentValue)}
                >
                  <SelectedCountry {...country} />
                </CommandItem>
              ))}
            </CommandGroup>
          </CommandList>
        </Command>
      </PopoverContent>
    </Popover>
  );
}

type SelectedCountryProps = Partial<CountryItem> & {
  showLabel?: boolean;
};
const SelectedCountry = ({
  displayName,
  flagIcon = '',
  direction,
  showLabel = true,
}: SelectedCountryProps) => (
  <div
    className={`${direction === 'rtl' ?? 'flex-row-reverse'} rtl:flex-row-reverse flex w-full justify-between gap-2 overflow-hidden items-center`}
  >
    {showLabel && <span className="text-xs text-black">{displayName}</span>}
    <div className="border rounded-sm overflow-hidden w-[24px] min-h-[18.5px] block">
      <img
        className="w-6"
        src={`https://cdn.jsdelivr.net/gh/lipis/flag-icons/flags/4x3/${flagIcon}.svg`}
        alt={displayName || ''}
      />
    </div>
  </div>
);

export { CountryItem, CountrySelector };
