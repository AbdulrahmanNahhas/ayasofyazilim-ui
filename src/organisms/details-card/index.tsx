import React from 'react';
// @ts-ignore

import { Button } from '@/components/ui/button';
import { IAboutCardProps } from '../../molecules/about-card';
import { ICardTableProps } from '../../molecules/card-table';
import { ICardTagProps } from '../../molecules/card-tag';
import Progress from '../../molecules/progress';
import Compact from './compact';
import FullPage from './full-page';

export interface IDetailsCardProps {
  ActionComponent?: React.ReactNode;
  BeforeCardContentComponent?: React.ReactNode;
  ContainerClassName?: string;
  IAboutCardProps: IAboutCardProps;
  cardTagClassname?: string;
  cardTagTitle?: string;
  cardTagVariant: ICardTagProps['variant'];
  description: string;
  image: string;
  link: string;
  locale?: string;
  tableProps?: Array<ICardTableProps>;
  tableProps2Col?: Array<[ICardTableProps, ICardTableProps]>;
  tags: Array<string>;
  title: string;
}
const currencyFormatter = new Intl.NumberFormat('tr', {
  style: 'currency',
  currency: 'TRY',
  maximumFractionDigits: 0,
});
export type DetailsProps = {
  cardProps: IDetailsCardProps;
  variant: 'compact' | 'full-page';
};
export const defaultCardProps: IDetailsCardProps = {
  IAboutCardProps: {
    link: 'https://google.com',
    avatar:
      'https://i.kickstarter.com/assets/043/950/483/f7c5bac8005024eea6c3ce6eaf65bb15_original.jpg?anim=false&fit=crop&height=80&origin=ugc&q=92&width=80&sig=IUCq8Z9OX16OY%2BmX17njzYURwPLYdY1ZcjVOuL%2FJfwc%3D',
    title: 'Clevetura Devices LLC',
    description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  link: 'https://google.com',
  title: 'CLVX 1 - Keyboard Gives More',
  description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  tags: ['Teknoloji', 'Yazılım'],
  image:
    'https://i.kickstarter.com/assets/044/243/111/620ee2c08af65e9646d6cfd9dbe55868_original.png?anim=false&fit=crop&gravity=faces&height=315&origin=ugc&q=92&width=560&sig=EXsDXlcpA3rTqgOLDiduLcetM6QIEzSCQh19YMy8nl4%3D',
  locale: 'tr',
  tableProps: [
    { title: 'Proje Tipi', value: 'Paya Dayalı' },
    { title: 'Pay Arz Oranı', value: '%8' },
    {
      title: 'Başlangıç Tarihi',
      value: new Date('01.02.2024').toLocaleDateString(),
    },
    {
      title: 'Bitiş Tarihi',
      value: new Date('01.03.2024').toLocaleDateString(),
    },
  ],
  tableProps2Col: [
    [
      {
        title: currencyFormatter.format(10000000).replace(/\s/g, ' '),
        value: 'Gerçekleşen Yatırım',
      },
      {
        title: currencyFormatter.format(10000000).replace(/\s/g, ' '),
        value: 'Hedeflenen Yatırım',
      },
    ],
  ],
  cardTagTitle: 'Başarılı',
  cardTagVariant: 'success',
  BeforeCardContentComponent: (
    <Progress value={20} containerClassName="h-3" className="bg-green-300" />
  ),
  ActionComponent: <Button>Yatırım yap</Button>,
};
export const defaultDetailsCardProps: DetailsProps = {
  variant: 'compact',
  cardProps: defaultCardProps,
};
export default function DetailsCard(infoCardProps: DetailsProps) {
  return infoCardProps.variant === 'full-page' ? (
    <FullPage {...infoCardProps.cardProps} />
  ) : (
    <Compact {...infoCardProps.cardProps} />
  );
}
