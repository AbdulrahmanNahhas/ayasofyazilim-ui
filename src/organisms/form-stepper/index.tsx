'use client';

import React from 'react';
import Stepper, { StepperContent } from '../stepper';
import AutoForm, { AutoFormProps } from '../auto-form';

const FormStepperGenerator = (steps: Step[]) =>
  steps.map((step: Step) => (
    <StepperContent title={step.title}>
      <AutoForm fieldConfig={{ withoutBorder: true }} {...step.autoformArgs} />
    </StepperContent>
  ));

export type Step = {
  autoformArgs: AutoFormProps;
  title: string;
};

export default function FormStepper({ steps }: { steps: Step[] }) {
  const [currentStep, setCurrentStep] = React.useState(0);
  const stepsComponents = FormStepperGenerator(steps);
  return (
    <Stepper
      activeTabIndex={currentStep}
      onIndexChange={setCurrentStep}
      vertical
    >
      {...stepsComponents}
    </Stepper>
  );
}
