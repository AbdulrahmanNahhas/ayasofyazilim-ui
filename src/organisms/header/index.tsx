import React from 'react';

import Navigation from '../../molecules/navigation-menu';
import { navigationLinkTypes } from '../../molecules/navigation-menu/types';
import AvatarWrapper from '../../molecules/avatar';
import { UserNav, userNavTypes } from '../profile-menu/index';
import BurgerMenu from '../burger-menu';

interface DashboardHeaderProps {
  children?: React.ReactNode;
  extraMenu?: React.ReactNode;
  logo?: string;
  navMenu: navigationLinkTypes[];
  title?: string;
  userNav?: userNavTypes;
}

export default function DashboardHeader({
  children,
  title,
  logo,
  userNav,
  navMenu,
  extraMenu,
}: DashboardHeaderProps) {
  return (
    <div className="flex items-center gap-4 h-12 px-2 w-100 shadow-sm">
      <BurgerMenu navigationLinks={navMenu} className="md:hidden" />
      <AvatarWrapper text="UR" url={logo} sideText={title} />
      <Navigation className="hidden md:flex" navigationLinks={navMenu} />
      {children && <div className="flex items-center gap-2">{children}</div>}
      <div className="flex items-center gap-2 ml-auto ">
        {extraMenu && <div>{extraMenu}</div>}
        <UserNav {...userNav} />
      </div>
    </div>
  );
}
