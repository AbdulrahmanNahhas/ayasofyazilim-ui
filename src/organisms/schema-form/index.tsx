import Form, { ThemeProps } from '@rjsf/core';
import { RJSFSchema } from '@rjsf/utils';
import validator from '@rjsf/validator-ajv8';
import { useState } from 'react';
import { AsyncSelect } from './fields/async-select';
import { CustomCheckbox } from './fields/checkbox';
import { CustomDate } from './fields/date';
import { FieldErrorTemplate } from './fields/error';
import { CustomPhoneField } from './fields/phone';
import { CustomSelect } from './fields/select';
import { CustomTextInput } from './fields/text';
import { AccordionArrayFieldTemplate } from './templates/array';
import { ErrorListTemplate } from './templates/error-list';
import { FieldTemplate } from './templates/field';
import { ObjectFieldTemplate } from './templates/object';
import { SchemaFormProps } from './types';
import {
  applyFiltersToSchema,
  flattenGenericData,
  generateFormData,
  generateUiSchema,
  hasPhoneFields,
  mergeUISchemaObjects,
  transformGenericSchema,
} from './utils';

const ShadcnTheme: ThemeProps = {
  fields: {
    phone: CustomPhoneField,
  },
  widgets: {
    CheckboxWidget: CustomCheckbox,
    SelectWidget: CustomSelect,
    'async-select': AsyncSelect,
    TextWidget: CustomTextInput,
    DateTimeWidget: CustomDate,
  },
  templates: {
    ArrayFieldTemplate: AccordionArrayFieldTemplate,
    ErrorListTemplate,
    FieldErrorTemplate,
    FieldTemplate,
    ObjectFieldTemplate,
  },
};

/**
 * SchemaForm component that renders a form based on the provided schema and options.
 * Extends the Form component from @rjsf/core.
 *
 * @param {SchemaFormProps} props - The props for the SchemaForm component.
 * @returns {JSX.Element} - The rendered form component.
 */
export function SchemaForm({ ...props }: SchemaFormProps) {
  const phoneFielsConfig = {
    fields: ['areaCode', 'ituCountryCode', 'localNumber'],
    requireds: ['areaCode', 'ituCountryCode', 'localNumber'],
    name: 'phone',
  };
  let uiSchema = {}; // Initialize the UI schema
  const { usePhoneField, filter } = props; // Start with the provided schema
  let { schema } = props;
  let statedForm = props.formData;
  // If the phone field is enabled, transform the schema and generate UI schema
  if (usePhoneField) {
    schema = transformGenericSchema(
      schema,
      phoneFielsConfig.fields, // Fields to transform
      phoneFielsConfig.name, // The parent field name
      phoneFielsConfig.requireds // Required fields
    );
    uiSchema = generateUiSchema(schema, phoneFielsConfig.name, {
      'ui:field': phoneFielsConfig.name, // Specify the field type for UI
    });
    if (hasPhoneFields(statedForm)) {
      statedForm = generateFormData(
        props.formData,
        phoneFielsConfig.fields, // Fields to transform
        phoneFielsConfig.name // The parent field name
      );
    }
  }
  if (filter) {
    schema = applyFiltersToSchema({ schema, filter });
  }
  // Merge any additional UI schema provided via props
  if (props.uiSchema) {
    uiSchema = mergeUISchemaObjects(uiSchema, props.uiSchema);
  }
  const [formData] = useState<any>(statedForm);
  return (
    <Form
      noHtml5Validate
      liveValidate
      focusOnFirstError
      showErrorList={props.showErrorList || false}
      {...props}
      formData={formData}
      schema={schema as RJSFSchema} // Cast schema to RJSFSchema type
      validator={validator} // Custom validator
      fields={{ ...ShadcnTheme.fields, ...props.fields }} // Merge custom fields
      widgets={{ ...ShadcnTheme.widgets, ...props.widgets }} // Merge custom widgets
      templates={{ ...ShadcnTheme.templates, ...props.templates }} // Merge custom templates
      uiSchema={uiSchema} // Set the generated UI schema
      onChange={(e) => {
        // Handle form data change
        if (props.usePhoneField) {
          e.formData = flattenGenericData(e.formData, 'phone', [
            'areaCode',
            'ituCountryCode',
            'localNumber',
          ]);
        }
        if (props.onChange) props.onChange(e); // Call the onChange prop if provided
      }}
      onSubmit={(_data, event) => {
        const data = _data;
        // Handle form submission
        if (props.usePhoneField) {
          data.formData = flattenGenericData(data.formData, 'phone', [
            'areaCode',
            'ituCountryCode',
            'localNumber',
          ]);
        }
        if (props.onSubmit) props.onSubmit(data, event); // Call the onSubmit prop if provided
      }}
    />
  );
}
