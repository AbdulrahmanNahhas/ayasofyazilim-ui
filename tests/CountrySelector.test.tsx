import React from 'react';
import renderer from 'react-test-renderer';
import { expect, it } from 'vitest';
import { CountrySelector } from '../src/organisms/country-selector';

it('renders correctly', () => {
  const tree = renderer
    .create(<CountrySelector menuAlign="end" defaultValue="tr" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
