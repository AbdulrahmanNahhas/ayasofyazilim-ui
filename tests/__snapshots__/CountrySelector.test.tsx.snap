// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<button
  aria-controls="radix-:r0:"
  aria-expanded={false}
  aria-haspopup="dialog"
  data-state="closed"
  onClick={[Function]}
  type="button"
>
  <div
    className="false rtl:flex-row-reverse flex w-full justify-between gap-2 overflow-hidden items-center"
  >
    <div
      className="border  rounded-sm overflow-hidden"
    >
      <img
        alt=""
        className="w-6"
        src="https://cdn.jsdelivr.net/gh/lipis/flag-icons/flags/4x3/.svg"
      />
    </div>
  </div>
</button>
`;
